week-5-practise/js/main.js

```diff
@@ -0,0 +1,88 @@
+//Query Select Buttons
+
+var blueButton = document.querySelector("#blueBtn");
+var orangeButton = document.querySelector("#orangeBtn");
+var greenButton = document.querySelector("#greenBtn");
+
+//Query Select elements
+
+var jumboTron = document.querySelector(".jumbotron");
+var donateBike = document.querySelector("#donateBikeBtn");
+var volunteer = document.querySelector(".btn-secondary");
+
+//Event Listener Functions
+
+blueButton.addEventListener("click", function() {
+  jumboTron.style.background = "#588fbd";
+  donateBike.style.background = "#ffa500";
+  volunteer.style.background = "black";
+  volunteer.style.color = "white";
+});
+
+orangeButton.addEventListener("click", function() {
+  jumboTron.style.background = "#f0ad4e";
+  donateBike.style.background = "#5751fd";
+  volunteer.style.background = "#31b0d5";
+  volunteer.style.color = "white";
+});
+
+greenButton.addEventListener("click", function() {
+  jumboTron.style.background = "#87ca8a";
+  donateBike.style.background = "black";
+  volunteer.style.background = "#8c9c08";
+});
+
+//Exercise 2
+
+
+//Query Select Form elements
+var emailInput = document.querySelector("#exampleInputEmail1");
+var nameInput = document.querySelector("#example-text-input");
+var describeInput = document.querySelector("#exampleTextarea");
+var submitBtn = document.querySelector("#submitButton");
+
+//Verification call-back functions
+function emailValid() {
+  return emailInput.value.includes("@") && emailInput.value.length > 0;
+}
+
+function nameValid() {
+  return nameInput.value.length > 0;
+}
+
+function describeValid() {
+  return describeInput.value.length > 0;
+}
+
+
+//Button Event Listener
+submitBtn.addEventListener(
+  "click",
```

Comment:
This will work, but we can actually be a bit smarter about how we listen for the form being submitted. One of the useful features of `<form>`s is that you can hit the Enter key to trigger the submission - you don't have to click  on the button.

This matters here because we are only listening for the `click` event on the submit button. Instead we really want to listen for the `submit` event that is first **on the form**. Then if we click a submit button or if we hit the Enter key, the event handler is fired 🎉.

Reply:
Hey Ali! I've pushed this again with your suggestions. Please let me know what you think!

For the eventListener on the button I have put two different options. I found that having the button event and then a key event seemed to be the way to get the form working the best. 

With the submit event, while it would submit the form on the enter key from within the form if you clicked outside of the form or you were in the describe text box hitting enter would do nothing.